includes/Core/Authentication/Authentication.php

```diff
@@ -1386,7 +1396,11 @@ private function filter_features_via_proxy( $feature_enabled, $feature_name ) {
 			return $feature_enabled;
 		}
 
-		$features = $this->get_transient_features();
+		if ( in_array( $feature_name, array( 'ideaHubModule', 'swgModule' ), true ) ) {
+			$features = $this->get_transient_features( $feature_name );
+		} else {
+			$features = $this->get_transient_features();
+		}
```

Comment:
See above, the entire "hacky" logic should be moved here, and we can leave `get_transient_features` as is then.